omod/src/main/java/org/openmrs/module/fhir2/narrative/OpenMRSNarrativeTemplate.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.narrative;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import ca.uhn.fhir.narrative2.INarrativeTemplate;
+import ca.uhn.fhir.narrative2.TemplateTypeEnum;
+import ca.uhn.fhir.rest.server.exceptions.InternalErrorException;
+import org.hl7.fhir.instance.model.api.IBase;
+
+public class OpenMRSNarrativeTemplate implements INarrativeTemplate {
```

Comment:
So my main question here is why we have this whole implementation when it seems like we could get away with this:

```java
public class OpenMRSNarrativeTemplate extends ca.uhn.fhir.narrative2.NarrativeTemplate {
    @Override
	public String getTemplateText() {
		try {
			return OpenMRSNarrativeTemplateManifest.loadResource(getTemplateFileName());
		} catch (IOException e) {
			throw new InternalErrorException(e);
		}
	}
}
```

Reply:
Actually doing this resulted in errors in `loadProperties()` method of `NarrativeTemplateManifest`, as it calls non-public methods of `NarrativeTemplate` like `setTemplateName()`, etc.

Reply:
I think that's what I'd like to see here. The idea is if we can make our implementation as light as possible, we're able to evolve with HAPI as easily as possible.